File: 241114/이상한 폭탄 2/strange-bomb-2.py
N, K = map(int, input().split())
arr = [int(input()) for _ in range(N)]

M = -1

for i in range(N):
    for j in range(i+1, N):
        if arr[j] == arr[i] and j-i <=K:
            M = max(M, arr[j])

print(M)

File: 250124/이상한 폭탄 2/strange-bomb-2.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    // Please write your code here.
    int arr[100];

    int N, K;
    cin >> N >> K;
    for (int i=0; i<N; i++) 
        cin >> arr[i];

    int max_val = -1;
    for (int i=0; i<N; i++) {
        for (int j=i+1; j<N; j++) {
            if (j-i > 3) continue;
            if (arr[i] == arr[j]) max_val = max(max_val, arr[i]);
        }
    }
    cout << max_val;
    
    return 0;
}